### PR TITLE
Login/History Servlets Testing

### PR DESCRIPTION
**Pull request by azhara-a**
**Write LoginServletTest and HistoryServletTest**
- Write getters for Recipe class
- Add constructors to LoginServlet and HistoryServlet
- Move getRecipeEntity() to RecipeCollector

### CODE REVIEW

**Review by mam249 — COMMENTED** [review]
LGTM

**Comment by mugdhalakhani** [comment]
> (Note: better to look at this once #40 is merged into this branch)
When building on an existing branch, you can set that as the base branch when creating your PR.

**Review by lminchillo — APPROVED** [review]
LGTM

File: backend/src/main/java/com/google/sps/data/Recipe.java [hunk 1]
```diff
@@ -65,6 +65,38 @@ public boolean containsIngredient(String ingredient) {
     return false;
   }
 
+  public String getName() {
```
Comment (mugdhalakhani) [inline backend/src/main/java/com/google/sps/data/Recipe.java:68]: nit: looks like these getters are only used for tests and so we should annotate with @VisibleForTesting

File: backend/src/main/java/com/google/sps/utils/RecipeCollector.java [hunk 2]
```diff
@@ -68,4 +70,41 @@
     }
     return recipes;
   }
+
+  public static Entity getRecipeEntity(JsonObject recipe) {
```
Comment (mugdhalakhani) [inline backend/src/main/java/com/google/sps/utils/RecipeCollector.java:74]: nit: namethis getRecipeEntityForTesting and annotate method with @VisibleForTesting

**Review by mugdhalakhani — APPROVED** [review]
LGTM, thanks for adding these!

File: backend/src/main/java/com/google/sps/utils/RecipeCollector.java (hunk 2 above)
Comment (azhara-a) [reply backend/src/main/java/com/google/sps/utils/RecipeCollector.java:74]: This method is not only used by tests, it is used in HistoryServlet class as well, so I don't think this applies to it.

**Pull request merged**